types.d.ts

```diff
@@ -0,0 +1,78 @@
+declare module "gcs-browser-upload" {
+  type ChunkUploadData = {
+    totalBytes: number
+    uploadedBytes: number
+    chunkIndex: number
+    chunkLength: number
+  }
+
+  type UploadOptions = {
+    /** A unique ID for the upload */
+    id: string
+
+    /** GCS resumable URL */
+    url: string
+
+    /** The File to upload */
+    file: File
+
+    /**
+     * Chunk size must be a multiple of 262144
+     * @default 262144
+     */
+    chunkSize?: number
+
+    /**
+     * Storage mechanism used to persist chunk metadata
+     * @default window.localStorage
+     */
+    storage?: Storage
+
+    /**
+     * Content type of the file being uploaded
+     * @default 'text/plain'
+     */
+    contentType?: string
+
+    /** A function that will be called with progress information */
+    onChunkUpload?: (data: ChunkUploadData) => void
+  }
+  export default class Upload {
+    static errors: typeof UploadErrors
+
+    constructor(config: UploadOptions, allowSmallChunks?: boolean)
+
+    start(): Promise<void>
```

Comment:
Ah you're right.

I have mixed feelings about this because this is exposing an implementation concern. The users of this API already can already catch and decipher errors & get progress using `onChunkUpload`, so I don't really see why they'd want to get the response object here.

But on the other hand as you say the Javascript _does_ return the Axios response object...

Reply:
i dont think anything is using the return value too

https://github.com/search?p=1&q=org%3Aqubitdigital+gcs-browser-upload&type=Code

maybe this PR and respective package version could include code to remove references to `lastResult`? then the type would be accurate